test-suite/src/test/scala/org/scalajs/testsuite/compiler/OptimizerTest.scala

```diff
@@ -81,6 +81,64 @@ object OptimizerTest extends JasmineTest {
 
   }
 
+  describe("String_+ constant folding") {
```

Comment:
The "public" name of `String_+` is actually `+[string]`. That is how it is printed by the IR printer. I'd rather you use that name here (and in the commit message).